src/processPullRequest.ts

```diff
@@ -37,14 +37,16 @@ export const processPullRequest = async (
 		throw new AggregateError(errs.map((v) => v.reason as Error));
 	}
 
-	const changeset = generateChangeset(
-		pr,
-		commits[0],
-		input,
-		changesetsConfig,
-		patch,
-		packageFiles.flatMap((v) => (v.status === 'fulfilled' ? [v.value] : [])),
-	);
+	const filterPrivatePackages
+		= typeof changesetsConfig.privatePackages === 'boolean'
+			? !changesetsConfig.privatePackages
+			: !((changesetsConfig.privatePackages?.tag ?? true) || (changesetsConfig.privatePackages?.version ?? true));
```

Comment:
I think you should only check for `changesetsConfig.privatePackages?.version` within the object, as it seems setting that to `false` hides the packages from Changesets CLI, not the `changesetsConfig.privatePackages?.tag` field.